### PR TITLE
Upgrade to prompt_toolkit 2.0

### PR DESCRIPTION
**Pull request by s-weigand**
This PR concerns #158 .
I tried to follow the changes made in[ ipython/ipython#11177](https://github.com/ipython/ipython/pull/11177). 
It passes the nosetests and works on linux (ubuntu18).
On Windows (gitbash + cmd tested) it only works, but the failed nosetests are the same as on master with the old ``prompt_toolkit `` so i hope its fine and I broke less than i repaired.

@takluyver Quite sneaky to present your tool (``enboard``) at PyData Berlin, knowing it breaks ppls Jupyter.
Thats how you get contributers ;)

P.S.: While i was in the code i did also some PEP8 changes.

### CODE REVIEW

**Comment by s-weigand** [comment]
Ahhh and the CI tests will ofc fail until the patched ipython version with [ ipython/ipython#11177](https://github.com/ipython/ipython/pull/11177) is released.
Except you want to hotfix it by installing ``pip install git+https://github.com/ipython/ipython.git`` on travis.

**Comment by takluyver** [comment]
Thanks, I've had a quick glance and I think this looks OK.

We generally ask people not to do style cleanups except on lines they're touching anyway, because it can cause unnecessary merge conflicts. But you've had to touch quite a few lines, and jupyter_console is quite quiet, so I don't mind letting it through in this case.

I think it's OK to install the IPython dev version on Travis for this change. We can open an issue to remind us to revert it once IPython 7 is out.

**Comment by s-weigand** [comment]
Maybe its i good idea to just open a feature branch with 7.x (as IPython does) and merge the PR there.
Since jupyter_console strongly depends on IPython, it would be reasonalbe to me to link the version numbers. Buuuuuut thats ofc up to you ^^

**Comment by dalejung** [comment]
@s-weigand so I had an issue where previous command entry was showing up as the default prompt. The following commit fixed it:

https://github.com/dalejung/jupyter_console/commit/02be37581ec5ead2e5a76121c7bbc7e7d74a6770

I'm just getting re-acclimated with the jupyter codebase so I have no idea if that's right-right.

**Comment by Carreau** [comment]
Thanks @dalejung I've added your commit to this PR, I've also unpinned IPython and installed it from the master branch of the main repo (which will become 7.0.0). 

Apologies for the delay to respond and review, this should get better after JupyterCon.

**Comment by Carreau** [comment]
Test are now passing. Though the updates requires us to drop support for Python 2.7 and 3.3. That should be pretty easy.

**Comment by Carreau** [comment]
I've updated `setup.py`  with `python_requires`, and extra check to avoid installation on Python 2, as well as not test on 3.3 and 2.7.

Test seem to pass. Would love an extra pair of eyes and the latests commits and get that in.

File: jupyter_console/ptshell.py [hunk 1]
```diff
@@ -512,8 +535,8 @@ def interact(self, display_banner=None):
             try:
                 code = self.prompt_for_code()
             except EOFError:
-                if (not self.confirm_exit) \
-                    or ask_yes_no('Do you really want to exit ([y]/n)?','y','n'):
+                if (not self.confirm_exit) or \
```
Comment (dsblank) [inline jupyter_console/ptshell.py]: Don't need a \ line continuation inside parens.

File: setup.py [hunk 2]
```diff
@@ -79,16 +98,19 @@
     'jupyter_client',
     'ipython',
     'ipykernel', # bless IPython kernel for now
-    'prompt_toolkit>=1.0.0,<2.0.0',
+    'prompt_toolkit>=2.0.0,<2.1.0',
```
Comment (dsblank) [inline setup.py]: Should we have an upper version number on prompt_toolkit? Is there something we know about 2.1.0?

Comment (s-weigand) [reply setup.py]: I basically just followed [ ipython/ipython#11177](https://github.com/ipython/ipython/pull/11177).
Maybe @takluyver (who made the `promt_toolkit 2` adoption for ipython), has some insigth on this.

File: jupyter_console/ptshell.py (hunk 1 above)
Comment (s-weigand) [reply jupyter_console/ptshell.py]: If i remove the "\" flake 8 tell me that there is a syntax error.
```
jupyter_console\ptshell.py:538:46: E999 SyntaxError: invalid syntax
jupyter_console\ptshell.py:539:25: E113 unexpected indentation
```

**Comment by Carreau** [comment]
> There are only minor issues pending; should we merge and continue working?

We can, IPython 7 is out so that should simplify working on that.

@mpacer you seemed to care about console want to merge and iterate ?

**Comment by mpacer** [comment]
This looks reasonable to me — let's merge and iterate!

**Comment by dsblank** [comment]
Thanks all!

**Comment by s3afroze** [comment]
pip install 'ipykernel<5.0.0'

A solution was posted here
https://github.com/jupyter/notebook/issues/4050

**Pull request merged**